NitroxClient/GameLogic/StorageSlots.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic.Helper;
+using NitroxClient.GameLogic.Spawning;
+using NitroxClient.Unity.Helper;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Helper;
+using NitroxModel.Logger;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic
+{
+    public class StorageSlots
+    {
+        private readonly IPacketSender packetSender;
+        private readonly LocalPlayer localPlayer;
+
+        public StorageSlots(IPacketSender packetSender, LocalPlayer localPlayer)
+        {
+            this.packetSender = packetSender;
+            this.localPlayer = localPlayer;
+        }
+
+
+        public void BroadcastItemAdd(InventoryItem item, GameObject gameObject)
+        {
+            string guid = GuidHelper.GetGuid(gameObject);
+            NitroxEntity entity = gameObject.GetComponent<NitroxEntity>();
+            if (entity == null)
+            {
+                return;
+            }
+
+            string itemGuid = GuidHelper.GetGuid(item.item.gameObject);
+            byte[] bytes = SerializationHelper.GetBytes(item.item.gameObject);
+
+            ItemData itemData = new ItemData(guid, itemGuid, bytes);
+            StorageSlotItemAdd add = new StorageSlotItemAdd(itemData);
+            packetSender.Send(add);
+        }
+
+        public void BroadcastItemRemoval(GameObject gameObject)
+        {
+            string guid = GuidHelper.GetGuid(gameObject);
+            NitroxEntity entity = gameObject.GetComponent<NitroxEntity>();
+            if (entity == null)
+            {
+                return;
+            }
+
+            StorageSlotItemRemove slotItemRemove = new StorageSlotItemRemove(guid);
+            packetSender.Send(slotItemRemove);
+        }
+
+        public void AddItem(GameObject item, string containerGuid, bool silent = false)
+        {
+            GameObject owner = GuidHelper.RequireObjectFrom(containerGuid);
```

Comment:
This should throw an exception if it does not get a game object back.  You don't have to check for null.

Reply:
It will anyway cause of the validate inside requireObject. I can leave that out.
Again i stole the code a bit from ItemContainer